prime-router/src/test/kotlin/SimpleReportTests.kt

```diff
@@ -151,13 +150,106 @@ class SimpleReportTests {
         compareTestResultsToExpectedResults(fakeReportFile, fakeReportFile2)
     }
 
-    private fun compareTestResultsToExpectedResults(testFile: File, expectedResultsFile: File) {
-        println("SimpleReportTests: diff'ing actual vs expected: ${testFile.absolutePath}    ${expectedResultsFile.absolutePath}")
+    private fun compareTestResultsToExpectedResults(
+        testFile: File,
+        expectedResultsFile: File,
+        compareKeys: Boolean = true,
+        compareLines: Boolean = true
+    ) {
         assertTrue(testFile.exists())
         assertTrue(expectedResultsFile.exists())
         // A bit of a hack:  diff the two files.
         val testFileLines = testFile.readLines()
         val expectedResultsLines = expectedResultsFile.readLines()
-        assertEquals(testFileLines, expectedResultsLines)
+
+        val testLines = convertFileToMap(testFileLines)
+        val expectedLines = convertFileToMap(expectedResultsLines)
+        val headerRow = expectedResultsLines[0].split(",")
+
+        if (compareKeys) {
+            // let's first compare the keys
+            compareKeysOfMaps(expectedLines, testLines)
+        }
+
+        if (compareLines) {
+            // let's compare our lines now
+            compareLinesOfMaps(expectedLines, testLines, headerRow)
+        }
+    }
+
+    private fun compareKeysOfMaps(expected: Map<String, Any?>, actual: Map<String, Any?>) {
+        val actualKeys = actual.keys.toSet()
+        val expectedKeys = expected.keys.toSet()
+
+        assertTrue(
+            actualKeys.minus(expectedKeys).count() == 0,
+            "There are keys in actual that are not in expected: ${actualKeys.minus(expectedKeys).joinToString { "," }}"
+        )
+        assertTrue(
+            expectedKeys.minus(actualKeys).count() == 0,
+            "There are keys in expected that are not present in actual: ${expectedKeys.minus(actualKeys).joinToString { "," }}"
+        )
+    }
+
+    private fun compareLinesOfMaps(
```

Comment:
Cool!   We could even move this into "regular" code, and expose it as a general utility.     Example:  our customers may want a simple way to programmatically compare and validate two CSVs - we could expose this via an API someday.

Reply:
I think that would be good as well. It'll be useful down the road.